packages/teams-js/test/public/sharing.spec.ts

```diff
@@ -23,8 +25,31 @@ describe('sharing_v1', () => {
     }
   });
 
+  Object.keys(FrameContexts)
+    .map(key => FrameContexts[key])
+    .forEach(frameContext => {
+      if (!allowedContext.includes(frameContext)) {
+        it(`should not allow calls from ${frameContext} context`, async () => {
+          await utils.initializeWithContext(frameContext);
+          const shareRequest: sharing.IShareRequest<sharing.IURLContent> = {
+            content: [
+              {
+                type: 'URL',
+                url: 'https://www.microsoft.com',
+                preview: true,
+                message: 'Test',
+              },
+            ],
+          };
+          expect(() => sharing.shareWebContent(shareRequest)).toThrowError(
```

Comment:
Maybe we can use the constant you made in the message? Using a template from Lakhveer's PR: 

```
 `This call is only allowed in following contexts: ${JSON.stringify(
            allowedContext,
          )}. Current context: "${context}".`,
```

Reply:
Done

Reply:
Done

Reply:
Can you please update the message to be "This call is only allowed in the following following contexts:" ?

Reply:
Actually, one further update. It would be useful to see in the test failure message which function is failing the test, e.g. "sharing.shareWebContent is only allowed in the following contexts:"

Reply:
I was thinking to categorize all the test functions in describe block so that we would have some nice clarity.
For e.g.,
` describe('should test settings.getSettings function', () => {
    const allowedContexts = [
      FrameContexts.content,
      FrameContexts.settings,
      FrameContexts.remove,
      FrameContexts.sidePanel,
    ];
    const expectedSettings: settings.Settings = {
      suggestedDisplayName: 'someSuggestedDisplayName',
      contentUrl: 'someContentUrl',
      websiteUrl: 'someWebsiteUrl',
      entityId: 'someEntityId',
    };

    Object.values(FrameContexts).forEach(context => {
      if (!allowedContexts.some(allowedContexts => allowedContexts == context)) {
        it(`should not allow calls from ${context} context`, async () => {
          await utils.initializeWithContext(context);
          expect(() => {
            settings.getSettings(settings => {
              expect(settings).toBe(expectedSettings);
            });
          }).toThrowError(
            `This call is only allowed in following contexts: ${JSON.stringify(
              allowedContexts,
            )}. Current context: "${context}".`,
          );
        });
      }
    });

    it('should successfully get settings', async () => {
      await utils.initializeWithContext('settings');
      settings.getSettings(settings => {
        expect(settings).toBe(expectedSettings);
      });

      const message = utils.findMessageByFunc('settings.getSettings');
      expect(message).not.toBeNull();

      utils.respondToMessage(message, expectedSettings);
    });
  });` 

Output: -
                                      should test settings.setValidityState function
@microsoft/teams-js:       √ should not allow calls from content context (58 ms)
@microsoft/teams-js:       √ should not allow calls from authentication context (4 ms)
@microsoft/teams-js:       √ should not allow calls from task context (3 ms)
@microsoft/teams-js:       √ should not allow calls from sidePanel context (2 ms)
@microsoft/teams-js:       √ should not allow calls from stage context (2 ms)
@microsoft/teams-js:       √ should not allow calls from meetingStage context (3 ms)
@microsoft/teams-js:       √ should successfully set validity state to true (3 ms)
@microsoft/teams-js:       √ should successfully set validity state to false (2 ms)

something on the above lines.

Reply:
I love this idea!